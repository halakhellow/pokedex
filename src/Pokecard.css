.Pokecard {
    display: flex;
    flex-direction: column;
    text-align: center;
    border-radius: 5px;
    background-color: rgba(255, 255, 255, 0.9);
    width: 18rem;
    height: 25rem;
    padding: 10px;
    margin: 1rem;
    box-shadow: 7px 10px 12px -5px rgba(0, 0, 0, 0.6);
}

.Pokecard-data {
    font-weight: 500;
    font-size: 1.1rem;
}

.Pokecard-img {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 13rem;
    background-color: rgb(0 191 143 / 20%);
    border-radius: 5px;
}

.Pokecard img {
    max-width: 100%;
}

.Pokecard img:hover {
    transform: scale(1.1);
    opacity: 0.9;
    transition: 1s;
}