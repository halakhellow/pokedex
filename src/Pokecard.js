import React, { Component } from "react";
import "./Pokecard.css";

function formatId(id) {
  id = id.toString();
  if (id.length == 1) return `00${id}`;
  if (id.length == 2) return `0${id}`;
  if (id.length == 3) return `${id}`;
}
class Pokecard extends Component {
  render() {
    let imgId = formatId(this.props.id),
      imgLink = `https://assets.pokemon.com/assets/cms2/img/pokedex/detail/${imgId}.png`;
    return (
      <div className="Pokecard">
        <h2 data-testid="title"> {this.props.name} </h2>
        <div className="Pokecard-img">
          <img src={imgLink} alt={this.props.name} />
        </div>
        <p className="Pokecard-data">Type : {this.props.type}</p>
        <span className="Pokecard-data">EXP : {this.props.experience} </span>
      </div>
    );
  }
}

export default Pokecard;
