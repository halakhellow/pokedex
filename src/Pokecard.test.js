import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Pokecard from "./Pokecard";

test("pokecard should have a pokemon name", () => {
  let { queryByTestId } = render(
    <Pokecard id={3} name="Pikachu" type="electric" experience={112} />
  );
  expect(queryByTestId("title")).toBeTruthy();
});

test("should match snapshot ", () => {
  let tree = renderer
    .create(<Pokecard id={3} name="Pikachu" type="electric" experience={112} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
