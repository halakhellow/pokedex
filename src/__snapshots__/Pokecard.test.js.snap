// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot  1`] = `
<div
  className="Pokecard"
>
  <h2
    data-testid="title"
  >
     
    Pikachu
     
  </h2>
  <div
    className="Pokecard-img"
  >
    <img
      alt="Pikachu"
      src="https://assets.pokemon.com/assets/cms2/img/pokedex/detail/003.png"
    />
  </div>
  <p
    className="Pokecard-data"
  >
    Type : 
    electric
  </p>
  <span
    className="Pokecard-data"
  >
    EXP : 
    112
     
  </span>
</div>
`;
